OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs

```diff
@@ -322,7 +329,13 @@ void RefreshServerList()
 					}
 				}
 
-				Game.RunAfterTick(() => RefreshServerListInner(games));
+				var lanGames = lanGameLocations.Select(bl => {
+						var gs = new GameServer(MiniYaml.FromString(bl.Data)[0].Value);
+						gs.Address = bl.Address.ToString().Split(':')[0] + ":" + gs.Address.Split(':')[1];
+						return gs;
+					}).Where(g => g.Id != Platform.CurrentProcessId).GroupBy(gs => gs.Address).Select(g => g.Last()).ToList();
```

Comment:
How hard would it be to do this filtering off the beacons rather than the GameServers?  That would more cleanly separate the "find games on the network" vs "turn those games into server entries" steps.